src/apps/workspaces/service/check_access.py

```diff
@@ -264,3 +265,10 @@ async def check_integrations_retrieve_access(self, applet_id: uuid.UUID, type: s
         )
         if not access:
             raise IntegrationsRetrieveAccessDenied(type=type, applet_id=applet_id)
+
+    async def check_integrations_delete_access(self, applet_id: uuid.UUID, type: str):
```

Comment:
You have checks for `create`, `retrieve` and `delete`, but they are all the same validation. Does it make sense to reduce to a single validation until we need more granular permission?